uber/model_checks.py

```diff
@@ -119,6 +119,13 @@ def full_name(attendee):
 
 @validation.Attendee
 @ignore_unassigned_and_placeholders
+def legal_name(attendee):
+    if attendee.legal_name_opt and attendee.full_name == attendee.legal_name_opt:
+        return 'When entering a legal name, it must be different than your preferred name. Otherwise, leave it blank.'
```

Comment:
Instead of returning an error in this case, I'd be inclined to just define a ``@presave_adjustment`` which does this automatically.  In general, anytime we have an error which tells the user exactly what they did wrong and explains the specific change they need to make which is definitely going to fix it, then we might as well just make that change ourselves.

The only reason to leave this error is if we think people are sometimes going to fill out this box with the same name as their first/last name but actually want to edit it to something else that's not blank after seeing this error.  This seems unlikely and the hassle to what I'm sure will be many many people who just fill it out unthinkingly is probably high.

On the other hand, if we hide the "Legal Name" field behind a checkbox (see my comment below), such that you didn't even see it unless you checked the box, then this error message probably makes more sense, and we should keep it.